tests/test_native_module.py

```diff
@@ -89,3 +89,30 @@ def test_distortion(gain_db, shape, sr=44100):
     np.testing.assert_equal(result.shape, full_scale_noise.shape)
     gain_scale = np.power(10.0, 0.05 * gain_db)
     np.testing.assert_allclose(np.tanh(full_scale_noise * gain_scale), result, rtol=4e-7, atol=2e-7)
+
+
+@pytest.mark.parametrize("reset", (True, False))
+def test_plugin_state_not_cleared_between_invocations(reset: bool):
+    """
+    Ensure that if `reset` is True, we do reset the plugin state
+    (i.e.: we cut off reverb tails). If `reset` is False, plugin
+    state should be maintained between calls to `render`
+    (preserving tails).
+    """
+    reverb = Reverb()
+    sr = 44100
+    noise = np.random.rand(sr)
+    silence = np.zeros_like(noise)
+
+    # Assert that reverb adds nothing if no signal was present already:
+    assert np.amax(np.abs(reverb(silence, sr, reset=reset))) == 0.0
```

Comment:
I'd consider making this check that the amax is < some small value like 1e-5 to avoid floating point comparison problems, but 0.0 shouldnt be an issue.